src/storage/src/source/mod.rs

```diff
@@ -982,15 +1525,34 @@ where
                     .cloned()
                     .unwrap_or(Timestamp::MAX),
             );
+
+            // It can happen that our view of the global source_upper is not yet
+            // up to date with what the ReclockOperator thinks. Ignore that for
+            // now.
```

Comment:
"Ignore that for now" means that this is something to be fixed in the code in the future or that the function will ignore it for now and it will do the correct thing in a future execution (and therefore the code needs no changes)?

Reply:
Updated the comment. The operator will eventually learn about new frontiers in future invocations.